### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by Mmduh-483**
Add section for SR-IOV Network Operator
Add section for system requirements

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -241,3 +242,15 @@ ib-sriov supports the following [CNI's Capabilities / Runtime Configuration](htt
...
+[SR-IOV Network Operator](https://github.com/openshift/sriov-network-operator) can manages the SR-IOV interfaces on the nodes e.g. change the number of VFs on the node, it can be used also to change the link type for the interfaces ETH to IB and vice versa, the [network policy example](https://github.com/openshift/sriov-network-operator/blob/master/deploy/crds/sriovnetwork.openshift.io_v1_sriovnetworknodepolicy_cr.yaml#L38) shows how to use the operator to change the link type for the given PCI Addresses of PFs.
+
+# System Requirements and Topology
+InfiniBand SR-IOV CNI works with kernel 5.5 which supports rdma namespace isolation and  read/write GUID of the VFs.
+The network topology for the InfiniBand SR-IOV CNI it is expected to has a node on the same network with OpenSM, the following diagram captures the InfiniBand SR-IOV topology for a two-node k8s cluster with external OpenSM node.
+
+![picture alt](docs/topology.png "Topology")
+
+Notes:
+* OpenSM should be from [Mellanox OFED](https://www.mellanox.com/products/infiniband-drivers/linux/mlnx_ofed) which can make VFs operational state up.
```
Comment (moshe010) [inline README.md:255]: s/OpenSM should be from/OpenSM with SR-IOV support should be download form/
remove the "which can make VFs operational state up"

File: README.md [hunk 2]
```diff
@@ -241,3 +242,15 @@ ib-sriov supports the following [CNI's Capabilities / Runtime Configuration](htt
...
+* OpnSM Node can be a worker node of the kubernetes cluster.
```
Comment (moshe010) [inline README.md:256]: s/OpnSM/OpenSM 
OpenSM can run on kubernetes master/worker node or on external server/switch

File: README.md [hunk 3]
```diff
@@ -241,3 +242,15 @@ ib-sriov supports the following [CNI's Capabilities / Runtime Configuration](htt
 
 EOF
 ```
+# SR-IOV Network Operator
+[SR-IOV Network Operator](https://github.com/openshift/sriov-network-operator) can manages the SR-IOV interfaces on the nodes e.g. change the number of VFs on the node, it can be used also to change the link type for the interfaces ETH to IB and vice versa, the [network policy example](https://github.com/openshift/sriov-network-operator/blob/master/deploy/crds/sriovnetwork.openshift.io_v1_sriovnetworknodepolicy_cr.yaml#L38) shows how to use the operator to change the link type for the given PCI Addresses of PFs.
+
+# System Requirements and Topology
+InfiniBand SR-IOV CNI works with kernel 5.5 which supports rdma namespace isolation and  read/write GUID of the VFs.
```
Comment (moshe010) [inline README.md:249]: extra  space and  read/

**Comment by moshe010** [comment]
do we have examples yaml folder?

**Comment by Mmduh-483** [comment]
> where is the multus crd configuration here?

There is already example https://github.com/Mellanox/ib-sriov-cni/blob/master/deployment/examples/ib-crd.yaml

**Comment by Mmduh-483** [comment]
> do we have examples yaml folder?

yes, https://github.com/Mellanox/ib-sriov-cni/tree/master/deployment/examples

File: README.md [hunk 4]
```diff
@@ -241,3 +242,15 @@ ib-sriov supports the following [CNI's Capabilities / Runtime Configuration](htt
 
 EOF
 ```
+# SR-IOV Network Operator
+[SR-IOV Network Operator](https://github.com/openshift/sriov-network-operator) can manages the SR-IOV interfaces on the nodes e.g. change the number of VFs on the node, it can be used also to change the link type for the interfaces ETH to IB and vice versa, the [network policy example](https://github.com/openshift/sriov-network-operator/blob/master/deploy/crds/sriovnetwork.openshift.io_v1_sriovnetworknodepolicy_cr.yaml#L38) shows how to use the operator to change the link type for the given PCI Addresses of PFs.
```
Comment (adrianchiris) [inline README.md:246]: can manages -> is used to manage

Comment (adrianchiris) [inline README.md:246]: >can manages the SR-IOV interfaces on the nodes 

is used to manage SR-IOV interfaces on the node

> it can be used also to change the link type for the interfaces ETH to IB and vice versa

it is also used to change the link type of an interface to Ethernet or Infiniband.

>change the link type for the given PCI Addresses of PFs
change the link type and SR-IOV attributes for a given PCI physical function address

File: README.md [hunk 5]
```diff
@@ -241,3 +242,15 @@ ib-sriov supports the following [CNI's Capabilities / Runtime Configuration](htt
...
+
+# System Requirements and Topology
+InfiniBand SR-IOV CNI works with kernel 5.5 which supports rdma namespace isolation and read/write GUID of the VFs.
```
Comment (adrianchiris) [inline README.md:249]: > with kernel 5.5
in RDMA CNI we require kernel 5.3 is there a reason to go with 5.5 ?

https://github.com/Mellanox/rdma-cni#overview

File: README.md [hunk 6]
```diff
@@ -241,3 +242,15 @@ ib-sriov supports the following [CNI's Capabilities / Runtime Configuration](htt
...
+The network topology for the InfiniBand SR-IOV CNI it is expected to has a node on the same network with OpenSM, the following diagram captures the InfiniBand SR-IOV topology for a two-node k8s cluster with external OpenSM node.
+
+![picture alt](docs/topology.png "Topology")
+
+Notes:
```
Comment (adrianchiris) [inline README.md:254]: id prefer a heading for notes i.e:

`##Notes`

File: README.md [hunk 7]
```diff
@@ -241,3 +242,15 @@ ib-sriov supports the following [CNI's Capabilities / Runtime Configuration](htt
 
 EOF
 ```
+# SR-IOV Network Operator
+[SR-IOV Network Operator](https://github.com/openshift/sriov-network-operator) can manages the SR-IOV interfaces on the nodes e.g. change the number of VFs on the node, it can be used also to change the link type for the interfaces ETH to IB and vice versa, the [network policy example](https://github.com/openshift/sriov-network-operator/blob/master/deploy/crds/sriovnetwork.openshift.io_v1_sriovnetworknodepolicy_cr.yaml#L38) shows how to use the operator to change the link type for the given PCI Addresses of PFs.
+
+# System Requirements and Topology
+InfiniBand SR-IOV CNI works with kernel 5.5 which supports rdma namespace isolation and read/write GUID of the VFs.
+The network topology for the InfiniBand SR-IOV CNI it is expected to has a node on the same network with OpenSM, the following diagram captures the InfiniBand SR-IOV topology for a two-node k8s cluster with external OpenSM node.
```
Comment (adrianchiris) [inline README.md:250]: I don't understand this sentence. What are you trying to convey here ?

File: README.md [hunk 8]
```diff
@@ -241,3 +242,15 @@ ib-sriov supports the following [CNI's Capabilities / Runtime Configuration](htt
...
+
+![picture alt](docs/topology.png "Topology")
```
Comment (adrianchiris) [inline README.md:252]: Some comments about the drawing.

- Worker nodes dont run kube API , they run  kubelet (or simply omit these services and just stick to kubernetes master/worker node marking)
- Switch symbols should be the same for ETH and IB, possibly with different colors and labels

File: README.md (hunk 5 above)
Comment (moshe010) [reply README.md:249]: What about the get Guid fix? I think it is kernel 5.5

File: README.md (hunk 7 above)
Comment (Mmduh-483) [reply README.md:250]: It is a description for the network topology, trying to explain the connections in the image

File: README.md (hunk 8 above)
Comment (Mmduh-483) [reply README.md:252]: Updated to the workers to kubelet and the symbol

**Comment by Mmduh-483** [comment]
@adrianchiris Updated the PR

File: README.md (hunk 5 above)
Comment (adrianchiris) [reply README.md:249]: yes 5.5 and the VF index fixed is in 5.6 as it was proposed just after 5.5 release.

File: README.md [hunk 9]
```diff
@@ -241,3 +242,15 @@ ib-sriov supports the following [CNI's Capabilities / Runtime Configuration](htt
 
 EOF
 ```
+# SR-IOV Network Operator
+[SR-IOV Network Operator](https://github.com/openshift/sriov-network-operator) is used to manage the SR-IOV interfaces on the nodes e.g. change the number of VFs on the node, it is also used to change the link type for the interfaces ETH to IB and vice versa, the [network policy example](https://github.com/openshift/sriov-network-operator/blob/master/deploy/crds/sriovnetwork.openshift.io_v1_sriovnetworknodepolicy_cr.yaml#L38) shows how to use the operator to change the link type and SR-IOV attributes for a given PCI physical function address.
+
+# System Requirements and Topology
+InfiniBand SR-IOV CNI works with kernel 5.5 which supports rdma namespace isolation and read/write GUID of the VFs.
```
Comment (adrianchiris) [inline README.md:249]: >InfiniBand SR-IOV CNI works with kernel 5.5 which supports rdma namespace isolation and read/write GUID of the VFs.

InfiniBand SR-IOV CNI works with kernel 5.5 which supports RDMA network namespace isolation and get/set of a VF's port and node GUID.

File: README.md (hunk 7 above)
Comment (adrianchiris) [reply README.md:250]: I would reword this sentence then.

The recommended network topology for a Kubernetes deployment with Infiniband as a secondary network is as follows:
- Two physical networks, one Ethernet network used as Kubernetes management and Pod primary network (these can be separate) and another Infiniband network interconnecting Kubernetes worker nodes
- Worker nodes where ib-sriov CNI is invoked are expected to have connectivity through the infiniband fabric with the subnet manager (SM), either running on a managed Infiniband Switch or another node.. (i.e there needs to be an active SM in the fabric)

File: README.md [hunk 10]
```diff
@@ -241,3 +242,18 @@ ib-sriov supports the following [CNI's Capabilities / Runtime Configuration](htt
...
+
+The recommended network topology for a Kubernetes deployment with Infiniband as a secondary network is as follows:
+* Two physical networks, one Ethernet network used as Kubernetes management and Pod primary network (these can be separate) and another Infiniband network interconnecting Kubernetes worker nodes.
+* Worker nodes where ib-sriov CNI is invoked are expected to have connectivity through the infiniband fabric with the subnet manager (SM), either running on a managed Infiniband switch or another node (i.e there needs to be an active SM in the fabric).
+
+![picture alt](docs/topology.png "Topology")
+
+## Notes:
+* OpenSM with SR-IOV support should be download form [Mellanox OFED](https://www.mellanox.com/products/infiniband-drivers/linux/mlnx_ofed).
+* OpenSM can run on kubernetes master/worker node or on external server/switch.
```
Comment (adrianchiris) [inline README.md:259]: This note is no longer needed

File: README.md (hunk 5 above)
Comment (moshe010) [reply README.md]: so please update to 5.6 kernel

Comment (adrianchiris) [reply README.md]: i dont think we use this functionality, as we GET the hardware address and SET the port and node guid so its not critical to update to 5.6 IMO

(issue that was fixed was the netlink GET port/node GUID of VF)
